pyscripts/srp.py

```diff
@@ -36,79 +55,70 @@ def __init__(self,K):
-            # I don't think that this will work
-            # I don't know how to implement waiting for all to have sent na in just one call sa start()
-            while(len(self.__malicious_sensors) + len(self.add_clean_sensor) < self.__K):
-                # wait for 1 minute and ask for sensorID and data again
-                pass
-            if len(self.__malicious_sensors >= self.__theta):
-                # announce manual investigation
-                # tell gateway to cache upcoming sensor data
-                self.send_to_gateway("START_MANUAL_INVESTIGATION")
-                investigation_result = None # this will be something
-                if investigation_result == "legit spike":
-                    # tell gateway to drop cached sensor data and restart model
-                    return "restart model"
-                elif investigation_result == "legit malicious":
-                    # tell gateway to remove malicious sensors from the cluster
-                    # and store cached data of clean sensors
-                    return ("remove then store",self.__malicious_sensors,self.__clean_sensors)
-                pass
-            else:
-                for id in self.__malicious_sensors:
-                    self.__malicious_sensors[id].trust_points -= 1
-                to_remove = filter(lambda x: x.trust_points == 0, self.__malicious_sensors)
-                # tells gateway to remove the sensors in the list called to_remove
-                return ("remove_from_cluster",to_remove)
\ No newline at end of file
+
+    def proceed_to_investigation(self):
+        if len(self.__malicious_sensors >= self.__theta):
+            # announce manual investigation
+            # tell gateway to cache upcoming sensor data
+            self.__message_to_gateway = "START_MANUAL_INVESTIGATION"
+            self.__message_from_gateway = None # the value of this should be changed sa gateway.py
+            if self.__message_from_gateway == "legit spike":
+                # tell gateway to drop cached sensor data and restart model
+                self.__message_to_gateway = "restart model"
+                return "restart model"
+            elif self.__message_from_gateway == "legit malicious":
+                # tell gateway to remove malicious sensors from the cluster
+                # and store cached data of clean sensors
+                self.__message_to_gateway = ("remove then store",self.__malicious_sensors,self.__clean_sensors)
+                return ("remove then store",self.__malicious_sensors,self.__clean_sensors)
+        else:
+            for id in self.__malicious_sensors:
+                self.__malicious_sensors[id].trust_points -= 1
+            to_remove = filter(lambda x: x.trust_points == 0, self.__malicious_sensors)
+            # tells gateway to remove the sensors in the list called to_remove
+            self.__message_to_gateway = ("remove_from_cluster",to_remove)
```

Comment:
Wrap `filter()` with `list()` to convert it to `list`. `filter()` returns a generator, not a list.